### PR TITLE
Spruce up dIBC implementation

### LINKED ISSUE #4153: endpoint not found error from ibcport connect()
## Describe the bug

Since I have one IBC connection in place...
```
command[80] E(conn).getRemoteAddress()
history[80] "/ibc-hop/connection-3/ibc-port/transfer/unordered/ics20-1/ibc-channel/channel-41"
```

@michaelfig suggested that I should be able to use that remote address less `/ibc-channel/channel-41` to make another connection, but It's not working:

```
command[86] E(home.ibcport[0]).connect(`/ibc-hop/connection-3/ibc-port/transfer/unordered/ics20-1`)
history[86] Promise.reject("Error: \"endpoint\" not found: (a string)")
```

## To Reproduce
1. `hermes create channel`
https://gist.github.com/dckc/9a336a983a30a20751b38380e3308fab#starting-a-hermes-relayer

_IOU; better file what I've got now before I run to this meeting..._

## Expected behavior

a connection

## Platform Environment

(lots of pieces)

 - what OS are you using? what version of Node.js?
 - is there anything special/unusual about your platform?
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)

## Screenshots
If applicable, add screenshots to help explain your problem, especially for UI interactions.

### PR DESCRIPTION
**Pull request by michaelfig**
refs: #4306 
closes: #4153

## Description

Modify the chain configuration so that the Pegasus implementation is moved from the `transfer` IBC port to the `pegasus` port.  Then move the `ibc-go` `transfer` implementation from the `cosmos-transfer` IBC port to the `transfer` port.

While testing this, I found there were several problems with dIBC.  This needed a dramatic overhaul and some new unit tests.

### Security Considerations

Still need to prevent Pegasus from accepting transfers to or from `vbank` assets, as is explained in #4306.

### Documentation Considerations

Will improve the usability of IBC transfer on the Agoric chain (works as expected by Cosmos ecosystem), so that Pegasus use is opt-in.

### Testing Considerations

Testing that `transfer` port on a local chain allows both sends and receives to another chain via standard (Hermes) tooling.

### CODE REVIEW

**Review by dckc — COMMENTED** [review]
I don't see a test... was that done manually? If so, perhaps I should reproduce it...

**Comment by michaelfig** [comment]
> I don't see a test... was that done manually? If so, perhaps I should reproduce it...

Yes, manual test with (I'm afk) I think Hermes 0.7.x.

**Comment by michaelfig** [comment]
Currently blocked (npi) by informalsystems/ibc-rs#1793.

File: packages/vats/src/ibc.js [hunk 1]
```diff
@@ -74,46 +71,45 @@ export function makeIBCProtocolHandler(E, rawCallIBCDevice) {
...
+   * @typedef {PromiseRecord<AttemptDescription>} OnConnectP
+   * @typedef {Omit<ConnectingInfo, 'counterparty' | 'channelID'> & {
+   *   localAddr: Endpoint, onConnectP: OnConnectP
+   *   counterparty: { port_id: string },
+   * }} Outbound
    */
 
   /**
-   * @type {Store<string, Array<Outbound>>}
+   * @type {LegacyMap<string, Array<Outbound>>}
```
Comment (dckc) [inline packages/vats/src/ibc.js:85]: I gather `LegacyMap` is motivated by mutability of the `Array`. Perhaps there's a sub-store arrangement that would work?

File: packages/vats/src/ibc.js [hunk 2]
```diff
@@ -302,9 +289,10 @@ export function makeIBCProtocolHandler(E, rawCallIBCDevice) {
       const onConnectP = makePromiseKit();
 
       pendingConns.add(onConnectP);
+      /** @type {Outbound[]} */
       let outbounds;
       if (srcPortToOutbounds.has(portID)) {
-        outbounds = srcPortToOutbounds.get(portID);
+        outbounds = [...srcPortToOutbounds.get(portID)];
```
Comment (dckc) [inline packages/vats/src/ibc.js:311]: @michaelfig tells me copying this array might not make sense

File: packages/vats/src/ibc.js (hunk 1 above)
Comment (michaelfig) [reply packages/vats/src/ibc.js:85]: Yes, IBC needs to be revisited once we do the SDK-wide revamp for durable/virtual collections.  I'm willing to postpone it until then.

File: packages/vats/src/ibc.js (hunk 2 above)
Comment (michaelfig) [reply packages/vats/src/ibc.js:311]: I'll redo this in the SDK-wide collections revamp.

**Review by dckc — APPROVED** [review]
We did a bunch of testing showing this works as designed, for example https://github.com/informalsystems/ibc-rs/pull/1801#issuecomment-1028093543

There seems to be a remaining hermes issue, but that need not hold back this PR.

Good stuff!

**Pull request merged**